testkit/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/testkit/PersistentEntitySpec.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.scaladsl.testkit
+
+import akka.actor.ActorSystem
+import com.lightbend.lagom.scaladsl.persistence.PersistentEntity
+import com.lightbend.lagom.scaladsl.playjson.JsonSerializerRegistry
+import com.lightbend.lagom.scaladsl.testkit.PersistentEntityTestDriver.Issue
+
+object PersistentEntitySpec {
+
+  /**
+    * An exception to be thrown if the issues after running a test are non-empty.
+    * @param entityId the entity which caused the issue.
+    * @param issues the sequence of issues
+    */
+  final case class PersistentEntityHasIssuesException(entityId: String, issues: Seq[Issue]) extends Exception {
+    override def getMessage: String = {
+      s"Persistent entity $entityId had ${issues.size} issue(s)...\nFirst issue: ${issues.head}"
+    }
+  }
+}
+
+trait PersistentEntitySpec {
+
+  type EntityCommand
+  type EntityEvent
+  type EntityState
+
+  /**
+    * The name of the actor system.
+    * @return
+    */
+  def actorSystemName: String
+
+  /**
+    * The registry required to set up the actor system.
+    * @return
+    */
+  def serializerRegistry: JsonSerializerRegistry
+
+  protected def actorSystem: ActorSystem = ActorSystem(
+    actorSystemName,
+    JsonSerializerRegistry.actorSystemSetupFor(serializerRegistry)
+  )
```

Comment:
I think the trait could _simply_ include a `def ActorSystem` and let users implement that bit so we push the responsibility to create/shutdown the actor system to user code. That is, I'd remove `actorSystemName`, serializerRegistry` and simplify `def actorSystem` to an abstract def.